report_gen/README.md

```diff
@@ -0,0 +1,187 @@
+
+|example/figs/graphgen/all:z4-x0.svg|example/figs/graphgen/all:z4-y1.svg|example/figs/graphgen/all:z4-y2.svg|example/figs/graphgen/all:z4-y3.svg|
+:-------------------------:|:-------------------------:|:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:z4-x0.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:z4-y1.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:z4-y2.svg" width="300" height="auto">|<img src="./example/figs/graphgen/all:z4-y3.svg" width="300" height="auto">
+
+#### Limit columns with pattern
+
+From the columns used in the basic use case, select the ones containing some pattern and plot using only those.
+
+The input patterns do not have to match with the columns perfectly; e.g. --plot_col y would plot columns y1,y2,y3.
+
+##### Sample command
+	python3 graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen --plot_col y1 z
+
+##### Sample output 
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-z4.svg|
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-z4.svg" width="300" height="auto"> 
+
+
+#### Output files extension
+
+Supported extensions are: "png", "jpeg", "webp", "svg", "pdf", "eps" and "html". 
+
+"html" creates an interactive plotly graph.
+
+##### Sample command
+    python3 graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen --plot_col y1 -e png svg html
+
+##### Sample output 
+|example/figs/graphgen/all:x0-y1.png|example/figs/graphgen/all:x0-y1.svg|
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y1.png" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto">
+
+**example/figs/graphgen/all:x0-y1.html**
+See example in example/figs/graphgen/all:x0-y1.html
+
+
+## Using htmlgen.py
+
+### Arguments: 
+    -f, --fig_dir - Path to the figures directory ( Required )
+    -t, --thumbnail_type - Extension of thumbnails on the main page ( Required )
+    -g, --graph_type - Extension of graphs ( Required )
+    -o, --out_dir - Output directory for the report ( Default: current dir )
+
+### Functionality
+
+Creates a html report from the graphs found in the figures directory. The report has 3 types of pages: the main page summarizing the individual tests by category, the pages for the individual tests themselves, and a page for comparing sensor outputs for all tests.
+
+All html pages are stored in a subdirectory of the output dir called htmls. An initial page called main.html is also stored in the output dir, which is just a redirect to the main page in the htmls directory.
```

Comment:
Why is not main.html called index.html. This is usually the default file looked up by web servers.